### ISSUE TITLE
Configuration on Demand should configure all parent projects

### ISSUE THREAD

**Issue by steffenyount** [gradle/gradle#11048]
Configuration on Demand presents users with a bit of a "footgun" scenario in that Gradle configurations found in the `allprojects {}`/`subprojects {}`/`ext {}` blocks of intermediate-parent/non-root projects may or may not be evaluated depending on what project configurations are demanded for a given build invocation. While the behavior is [well documented](https://docs.gradle.org/5.6.2/userguide/multi_project_builds.html#sec:configuration_on_demand), it still results in unnecessarily surprising behaviors and gotcha-bugs that undermine Gradle's reliability for the uninitiated user.

### Expected Behavior
In addition to always configuring the root-project, `Configuration on Demand` should also always configure all ancestor projects that lie between a project targeted for configuration and the root-project.

### Current Behavior
Only the root-project and other targeted projects are configured.

### Context
It's been a recurring internal dev-support issue affecting new projects added into our large multi-project repo, where devs authoring these new projects either don't know to (or forget to) add the necessary `evaluationDependsOn()` declarations so that their intermediate parent projects containing `allprojects {}`/`subprojects {}`/`ext {}` blocks do always get evaluated before their descendant projects.

**Comment by eskatos** [comment]
This would also make configuration on demand follow the classloader hierarchy, alleviating a whole class of problems. For example, currently the Gradle Kotlin DSL shouldn't be used together with configure on demand because of that.

This would also mean deprecating/removing the contradictory `evaluationDependsOnChildren()`.

**Comment by LouisCAD** [comment]
For which version of Gradle is this planned if at all?

**Comment by LouisCAD** [comment]
@eskatos Will a fix for this issue ever happen?

**Comment by bigdaz** [comment]
I think that this issue was fixed by https://github.com/gradle/gradle/pull/14616. @LouisCAD can you please check to see if this issue exists for you in Gradle 6.8.3?
cc @steffenyount @ghale

**Comment by steffenyount** [comment]
> I think that this issue was fixed by #14616. @LouisCAD can you please check to see if this issue exists for you in Gradle 6.8.3?
> cc @steffenyount @ghale

It does appear that this issue has been fixed!

I assembled a little test multi-project that I ran with v5.6.4 vs v6.8.3 looking like:
```
include 'sub1'
include 'sub1:sub1.1'
include 'sub1:sub1.1:sub1.1.1'
include 'sub1:sub1.2'

include 'sub2'
```
```
./gradlew -p sub1/sub1.1/sub1.1.1 printMyList
```
I got v5.6.4 results:
```
[root, sub1.1.1]
```
vs v6.8.3 results:
```
[root, sub1, sub1.1, sub1.1.1]
```


[tst-proj.zip](https://github.com/gradle/gradle/files/6159688/tst-proj.zip)

**Comment by jjohannes** [comment]
Closing as this was fixed in 6.8.3
